### PR TITLE
Backend and frontend changes to test BB groups

### PR DESCRIPTION
**Pull request by marcospri**

### CODE REVIEW

File: lms/resources/_js_config/__init__.py [hunk 1]
```diff
@@ -215,6 +215,12 @@ def enable_content_item_selection_mode(self, form_action, form_fields):
                     "formAction": form_action,
                     "formFields": form_fields,
                     "createAssignmentAPI": self._create_assignment_api(),
+                    # The "content item selection" that we submit to Canvas's
+                    # content_item_return_url is actually an LTI launch URL with
+                    # the selected document URL or file_id as a query parameter. To
+                    # construct these launch URLs our JavaScript code needs the
+                    # base URL of our LTI launch endpoint.
+                    "ltiLaunchUrl": self._request.route_url("lti_launches"),
```
Comment (marcospri) [inline lms/resources/_js_config/__init__.py:223]: This shouldn't be necessary when this is addressed:

https://github.com/hypothesis/lms/pull/3356

File: lms/resources/_js_config/__init__.py [hunk 2]
```diff
@@ -413,6 +419,7 @@ def _get_students(self):
             )
             yield {
                 "userid": h_user.userid(self._authority),
+                "lmsId": grading_info.user_id,
```
Comment (marcospri) [inline lms/resources/_js_config/__init__.py:422]: Split into its own PR https://github.com/hypothesis/lms/pull/3451

File: lms/services/group_info.py [hunk 3]
```diff
@@ -19,6 +19,7 @@ class GroupInfoService:
         "course": "course_group",
         "canvas_section": "section_group",
         "canvas_group": "canvas_group_group",
+        "blackboard_group": "blackboard_group_group",
```
Comment (marcospri) [inline lms/services/group_info.py:22]: On it's own PR https://github.com/hypothesis/lms/pull/3450

File: lms/resources/_js_config/file_picker_config.py [hunk 4]
```diff
@@ -31,6 +33,9 @@ def blackboard_config(cls, context, request, application_instance):
     def canvas_config(cls, context, request, application_instance):
         """Get Canvas files config."""
 
+        if not context.is_canvas:
+            return {}
```
Comment (marcospri) [inline lms/resources/_js_config/file_picker_config.py:37]: Related to https://github.com/hypothesis/lms/pull/3356 but if returning None here it breaks this branch.

File: lms/resources/_js_config/file_picker_config.py [hunk 5]
```diff
@@ -4,6 +4,8 @@ class FilePickerConfig:
     @classmethod
     def blackboard_config(cls, context, request, application_instance):
         """Get Blackboard files config."""
+        if not application_instance.product == application_instance.Product.BLACKBOARD:
+            return {}
```
Comment (marcospri) [inline lms/resources/_js_config/file_picker_config.py:8]: Related to https://github.com/hypothesis/lms/pull/3356 but if returning None here it breaks this branch.

File: lms/static/scripts/frontend_apps/components/LMSGrader.js [hunk 6]
```diff
@@ -57,9 +73,22 @@ export default function LMSGrader({
...
-    [clientRPC]
+    async (
+      /** @param {StudentInfo|null} user - The user to focus on in the sidebar */
+      user
+    ) => {
+      if (apiSync && user) {
+        const groups = await apiCall({
+          authToken,
+          path: apiSync.path,
+          data: { ...apiSync.data, gradingStudentId: user.lmsId },
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/components/LMSGrader.js:84]: @marcospri Ah, I see that `lmsId` was added a couple of days ago on the backend:

https://github.com/hypothesis/lms/commit/dcefb1e3eeda913b6c6605d4c4beec8e4a83e48f

I'll need to update the front-end typing to correspond.

Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/LMSGrader.js:84]: Actually, can you help me understand what the `lmsId` _is_, @marcospri ? And how it differs from `userid`?

Comment (marcospri) [reply lms/static/scripts/frontend_apps/components/LMSGrader.js:84]: The existing `userid` is the user on H while lmsId is the ID given to the user/student on the LMS (Blackboard here). We need the later to query for its groups on the group set while grading.
